// cypress-browser-permissions/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PERMISSIONS_PREF_NAME_BY_FAMILY = exports.PERMISSIONS_PREF_CONTAINER_BY_FAMILY = exports.PREFERENCES_ROOT_PATH_BY_FAMILY = exports.PLUGIN_ENV_VAR = void 0;
exports.PLUGIN_ENV_VAR = 'browserPermissions';
exports.PREFERENCES_ROOT_PATH_BY_FAMILY = {
    chromium: 'preferences.default',
    firefox: 'preferences',
};
exports.PERMISSIONS_PREF_CONTAINER_BY_FAMILY = {
    chromium: 'profile.managed_default_content_settings',
    firefox: 'permissions.default',
};
exports.PERMISSIONS_PREF_NAME_BY_FAMILY = {
    chromium: {
        notifications: 'notifications',
        geolocation: 'geolocation',
        images: 'images',
        cookies: 'cookies',
        javascript: 'javascript',
        plugins: 'plugins',
        camera: 'media_stream',
        microphone: 'media_stream',
        popups: 'popups',
    },
    firefox: {
        camera: 'camera',
        microphone: 'microphone',
        notifications: 'desktop-notification',
        geolocation: 'geo',
        images: 'image',
        cookies: undefined,
        javascript: undefined,
        plugins: undefined,
        popups: undefined,
    },
};
