// cypress-browser-permissions/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isPermission = exports.isPermissionAsk = exports.isPermissionBlocked = exports.isPermissionAllowed = exports.getBrowserPermission = exports.cypressBrowserPermissionsPlugin = void 0;
const constants_1 = require("./constants");
const types_1 = require("./types");
const helpers_1 = require("./helpers");
var plugin_1 = require("./plugin");
Object.defineProperty(exports, "cypressBrowserPermissionsPlugin", { enumerable: true, get: function () { return plugin_1.cypressBrowserPermissionsPlugin; } });
function getBrowserPermission(permission) {
    const permissions = Cypress.env(constants_1.PLUGIN_ENV_VAR);
    if (permissions && typeof permissions[permission] !== 'undefined') {
        return (0, helpers_1.toPermissionState)(permissions[permission]);
    }
    else {
        return undefined;
    }
}
exports.getBrowserPermission = getBrowserPermission;
function isPermissionAllowed(permission) {
    return getBrowserPermission(permission) === types_1.PermissionState.allow;
}
exports.isPermissionAllowed = isPermissionAllowed;
function isPermissionBlocked(permission) {
    return getBrowserPermission(permission) === types_1.PermissionState.block;
}
exports.isPermissionBlocked = isPermissionBlocked;
function isPermissionAsk(permission) {
    return getBrowserPermission(permission) === types_1.PermissionState.ask;
}
exports.isPermissionAsk = isPermissionAsk;
function isPermission(permission, state) {
    return getBrowserPermission(permission) === state;
}
exports.isPermission = isPermission;
